Criterion/Types.hs

```diff
@@ -660,6 +661,9 @@ instance Binary OutlierEffect where
             _ -> fail $ "get for OutlierEffect: unexpected " ++ show i
 instance NFData OutlierEffect
 
+instance Semigroup Outliers where
+    (<>) = addOutliers
+
 instance Monoid Outliers where
     mempty  = Outliers 0 0 0 0 0
     mappend = addOutliers
```

Comment:
While you're at it, can you change this to:

```haskell
#if !(MIN_VERSION_base(4,11,0))
     mappend = (<>)
#endif
```

This will help placate `-Wcompat`, and will make it forward compatible when we eventually drop `mappend` from `Monoid`.